Update pysam recipe and patch cython run dependency

mencian opened the pull request:
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

mencian commented:
@BiocondaBot please add label

jmarshall commented on recipes/pysam/build.sh:
```diff
@@ -3,15 +3,15 @@
 platform=$(uname)
-if [[ "$platform" == "Linux" ]]; then
-    export CFLAGS="-fPIC $CFLAGS"
+if [[ "${platform}" == "Linux" ]]; then
```
What is the reason for adding `{ … }` everywhere?

IMHO all it does is add noise in the git history. Please don't.

jmarshall commented on recipes/pysam/meta.yaml:
```diff
@@ -1,26 +1,31 @@
+{% set name = "pysam" %}
```
Please don't extract `name` and `sha256` like this. It's pointless.

jmarshall commented on recipes/pysam/meta.yaml:
```diff
@@ -1,26 +1,31 @@
-  binary_relocation: False # [linux]
+  number: 2
+  skip: True  # [py < 36]
+  binary_relocation: False  # [linux]
```
Please don't change lines without good reason. It just adds noise in the git history.

jmarshall commented on recipes/pysam/meta.yaml:
```diff
@@ -29,17 +34,19 @@ requirements:
 about:
-  home: https://github.com/pysam-developers/pysam
-  license: MIT
+  home: https://pypi.org/project/pysam/
```
What is the reason for updating `home` away from the place where the upstream pysam maintainers work?

jmarshall commented:
What is the reason for changing the build from `$PYTHON setup.py …` to `pip …`?

mencian commented:
I used grayskull to generate a template for pysam, which uses `pip install` by default, then added a few changes of my own. I have removed all the unecessary changes and cleaned up redundant run dependencies in my recent commit.
